crates/codegen/parser/runtime/src/napi_interface/query.rs

```diff
@@ -0,0 +1,92 @@
+// NAPI-exposed functions have to accept owned values
+#![allow(clippy::needless_pass_by_value)]
+// The functions are meant to be definitions for export, so they're not really used
+#![allow(clippy::return_self_not_must_use)]
+
+use napi::{Env, JsObject};
+use napi_derive::napi;
+
+use crate::napi_interface::cursor::Cursor;
+use crate::napi_interface::{RustQuery, RustQueryResultIterator};
+
+#[napi(namespace = "query")]
+pub struct Query(RustQuery);
```

Comment:
Not related to this PR, but in a future refactoring, I would like to use `struct Foo {}` for Rust types, and `#[napi(name = "Foo")] struct NapiFoo{}` for NAPI types, as I'm (and RA) are often confused by both, importing/using the wrong one Having two distinct type names (and no renames/re-exports) should fix it.

Just TOL.

Reply:
100% agree